src/site/content/en/blog/module-workers/index.md

```diff
@@ -0,0 +1,229 @@
+a Worker and composing scripts has remained similar to the synchronous script loading approaches
+common in 2009.
+
+## History: Classic Workers
+
+The Worker constructor takes a [Classic
+Script](https://html.spec.whatwg.org/multipage/webappapis.html#classic-script) URL, which is
+relative to the document URL. It immediately returns a reference to the new Worker instance, which
+exposes a messaging interface as well as a `terminate()` method that immediately stops and destroys
+the worker.
+
+```js
+const worker = new Worker('worker.js');
+```
+
+An `importScripts()` function is available within Web Workers for loading additional code, but it
+pauses execution of the worker in order to fetch and evaluate each script. It also executes scripts
+in the global scope like a classic `<script>` tag, meaning the variables in one script can be
+overwritten by the variables in another.
+
+##### worker.js:
+
+```js
+importScripts('greet.js');
+// ^ could block for seconds
+addEventListener('message', e => {
+  postMessage(sayHello());
+});
+```
+
+##### greet.js:
+
+```js
+// global to the whole worker
+function sayHello() {
+  return 'world';
+}
+```
+
+For this reason, using Web Workers has historically imposed an outsized effect on the architecture
+of an application. Developers have had to create clever tooling and workarounds to make it possible
+to use Web Workers without giving up modern development practises. As an example, bundlers like
+Webpack embed a small module loader implementation into generated code that uses `importScripts`
+for code loading, but wraps modules in functions to avoid variable collisions and simulate
+dependency imports and exports.
+
+
+## Enter Module Workers
+
+A new mode for Web Workers with the ergonomics and performance benefits of [JavaScript
+Modules](https://v8.dev/features/modules) is shipping in Chrome 80, called Module Workers. The
+Worker constructor accepts a new `{type:"module"}` option, which changes script loading and
+execution to match `<script type="module">`.
+
+```js
+const worker = new Worker('worker.js', {
+  type: 'module'
+});
+```
+
+Since Module Workers are JS Modules, they can use standard JavaScript import and export statements.
+As with all JS Modules, dependencies are only executed once in a given context (main thread,
+worker, etc), and all future imports reference the already-executed module instance. The loading
+and execution of JS Modules is also optimized by browsers - because a Module's dependencies can be
+loaded prior to the Module being executed, entire module trees can be loaded in parallel. Module
+loading also caches parsed code, which means modules that are used on the main thread and in a
+Worker only need to be parsed once.
+
+Moving to JS Modules also enables the use of [dynamic
+import](https://v8.dev/features/dynamic-import) for lazy-loading code without blocking execution of
+the worker. Dynamic Import is much more explicit than using importScripts to load dependencies,
```

Comment:
* 'Import' to 'import' (lower case)
* importScripts needs to be `importScripts()` (formatting and parens added)